### PR TITLE
feat(postprocessing): embeddable lua scripts for postprocessing

### PR DESCRIPTION
**Pull request by besaleli**
Big feature!

Lua scripts are embedded at compile time to handle embedding/logit postprocessing. Examples can be found in `transforms/` folder. This will be extended to custom pooling for sentence transformers.

### CODE REVIEW

**Review by javiermtorres — APPROVED** [review]
LGTM. I don't (can't) see any obvious issues. Just a couple of future improvements and questions.
Do you see any need for a more complicated example, for example chaining postprocessing steps?

File: encoderfile/src/transforms/tensor/mod.rs [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        methods.add_method("map_axis", |_, this, (axis, func)| {
+            this.map_axis(axis, func)
+        });
+        methods.add_method("fold_axis", |_, this, (axis, acc, func)| {
+            this.fold_axis(axis, acc, func)
+        });
+    }
+}
+
+impl Tensor {
```
[the remark this answers is not in the document]
Comment (besaleli) [reply encoderfile/src/transforms/tensor/mod.rs]: Yes, alongside scalar numbers in vector ops

File: encoderfile/src/inference/token_classification.rs [hunk 2]
```diff
@@ -72,12 +73,13 @@ pub fn postprocess(
                 },
                 score,
                 label,
+                // TODO: we only need to return one of these
```
[the remark this answers is not in the document]
Comment (besaleli) [reply encoderfile/src/inference/token_classification.rs]: This was from when we were including both raw logits and softmax-ed scores. now that we're including postprocessing, i think we should just send the postprocessed scores

File: encoderfile/src/transforms/tensor/mod.rs [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+use mlua::prelude::*;
+use ndarray::{Array1, ArrayD, Axis};
+use ndarray_stats::QuantileExt;
+use ort::tensor::ArrayExtensions;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Tensor(pub ArrayD<f32>);
```
[the remark this answers is not in the document]
Comment (besaleli) [reply encoderfile/src/transforms/tensor/mod.rs]: not for now—we'll deal with int8 models later
